packages/net/http/agent.pony

```diff
@@ -0,0 +1,69 @@
+use "collections"
+
+class iso Agent
+  """
+  Manages a client request payload.
+  """
+  let _client: Client
+  var _request: Payload
+
+  new iso create(method: String, url: URL, client: Client) =>
+    """
+    Create an agent for the given HTTP method, URL, and client.
+    """
+    _client = client
+    _request = Payload.request(method, url)
+
+  fun iso headers
```

Comment:
Calling `headers()` consumes the `Agent` which essentially means you can either inspect headers or send a request but not both. I was wondering what is the use case for it? Maybe a new `HashMap` with headers copied from `_request` would make more sense?

The same applies to `body()` and `body_size()`.